src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2200,13 +2200,22 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
 
 #if defined(FEATURE_EH_WINDOWS_X86)
         case GT_END_LFIN:
+        {
+            // Find the eh table entry via the eh ID
+            //
+            unsigned const ehID = (unsigned)treeNode->AsVal()->gtVal1;
```

Comment:
Could we just store `EHBlkDsc*` directly in `gtVal1` and avoid the lookup table?

Reply:
Or perhaps these pointers won't be stable after #112998?